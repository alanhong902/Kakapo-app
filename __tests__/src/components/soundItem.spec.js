import React from 'react';
import { map } from 'ramda';
import { newSoundObj } from 'utils/';
import { SoundItem } from 'components/';
import { getData, createComponentWithIntl } from '../helper';

function setup(props = {}) {
  const propData = {
    ...getData('themes'),
    soundActions: {},
    ...props,
  };
  return {
    props,
    tree: createComponentWithIntl(<SoundItem {...propData} />).toJSON(),
  };
}

const soundProp = (props = {}) => {
  const obj = { ...newSoundObj, source: 'file', progress: 1, ...props };
  return { sound: map(x => (x === null ? 'wind' : x), obj) };
};

test('<SoundItem/> render', () => {
  const { tree } = setup(soundProp());
  expect(tree).toMatchSnapshot();
});

test('<SoundItem/> w/o image', () => {
  const { tree } = setup(soundProp({ source: 'youtubeStream', img: '' }));
  expect(tree).toMatchSnapshot();
});

test('<SoundItem/> youtube render', () => {
  const { tree } = setup(soundProp({ source: 'youtubeStream' }));
  expect(tree).toMatchSnapshot();
});

test('<SoundItem/> youtube render test-id', () => {
  const { tree } = setup(
    soundProp({ file: '7ccQyyCLtx8', source: 'youtubeStream' })
  );
  expect(tree).toMatchSnapshot();
});

test('<SoundItem/> render playing', () => {
  const { tree } = setup(soundProp({ playing: true }));
  expect(tree).toMatchSnapshot();
});
