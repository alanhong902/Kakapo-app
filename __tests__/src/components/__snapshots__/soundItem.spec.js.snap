// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SoundItem/> render 1`] = `
<div
  className="item waves-effect waves-block paused"
  onClick={[Function]}
  role="button"
  style={
    Object {
      "color": "#121212",
    }
  }
  tabIndex={-1}
>
  <div
    className="inner"
  >
    <i
      className="preview icon-wind"
    />
    <div
      className="actions dark"
    >
      <div>
        <a
          href="wind"
          onClick={[Function]}
          rel="noopener noreferrer"
          target="_blank"
        >
          <i
            className="icon-share"
          />
        </a>
      </div>
      <div
        onClick={[Function]}
        role="presentation"
      >
        <i
          className="icon-edit"
        />
      </div>
      <div
        onClick={[Function]}
        role="presentation"
      >
        <i
          className="icon-delete"
        />
      </div>
    </div>
    <span
      className="title"
    >
      wind
    </span>
    <input
      defaultValue={0.5}
      max="1"
      min="0"
      onChange={[Function]}
      onClick={[Function]}
      step="0.01"
      type="range"
    />
  </div>
  <div />
</div>
`;

exports[`<SoundItem/> render playing 1`] = `
<div
  className="item waves-effect waves-block playing"
  onClick={[Function]}
  role="button"
  style={
    Object {
      "backgroundColor": "#673AB7",
      "color": "#fff",
    }
  }
  tabIndex={-1}
>
  <div
    className="inner"
  >
    <i
      className="preview icon-wind"
    />
    <div
      className="actions"
    >
      <div>
        <a
          href="wind"
          onClick={[Function]}
          rel="noopener noreferrer"
          target="_blank"
        >
          <i
            className="icon-share"
          />
        </a>
      </div>
      <div
        onClick={[Function]}
        role="presentation"
      >
        <i
          className="icon-edit"
        />
      </div>
      <div
        onClick={[Function]}
        role="presentation"
      >
        <i
          className="icon-delete"
        />
      </div>
    </div>
    <span
      className="title"
    >
      wind
    </span>
    <input
      defaultValue={0.5}
      max="1"
      min="0"
      onChange={[Function]}
      onClick={[Function]}
      step="0.01"
      type="range"
    />
  </div>
  <div />
</div>
`;

exports[`<SoundItem/> w/o image 1`] = `
<div
  className="item waves-effect waves-block paused youtube-stream"
  onClick={[Function]}
  role="button"
  style={
    Object {
      "color": "#121212",
    }
  }
  tabIndex={-1}
>
  <div
    className="inner"
  >
    <div
      className="no-image"
    />
    <div
      className="actions dark"
    >
      <div>
        <a
          href="wind"
          onClick={[Function]}
          rel="noopener noreferrer"
          target="_blank"
        >
          <i
            className="icon-share"
          />
        </a>
      </div>
      
      <div
        onClick={[Function]}
        role="presentation"
      >
        <i
          className="icon-delete"
        />
      </div>
    </div>
    <span
      className="title"
    >
      wind
    </span>
    <input
      defaultValue={0.5}
      max="1"
      min="0"
      onChange={[Function]}
      onClick={[Function]}
      step="0.01"
      type="range"
    />
  </div>
  <div
    className="youtube-video"
    id="video-wind"
  />
</div>
`;

exports[`<SoundItem/> youtube render 1`] = `
<div
  className="item waves-effect waves-block paused youtube-stream"
  onClick={[Function]}
  role="button"
  style={
    Object {
      "color": "#121212",
    }
  }
  tabIndex={-1}
>
  <div
    className="inner"
  >
    <img
      alt="wind"
      src="wind"
    />
    <div
      className="actions dark"
    >
      <div>
        <a
          href="wind"
          onClick={[Function]}
          rel="noopener noreferrer"
          target="_blank"
        >
          <i
            className="icon-share"
          />
        </a>
      </div>
      
      <div
        onClick={[Function]}
        role="presentation"
      >
        <i
          className="icon-delete"
        />
      </div>
    </div>
    <span
      className="title"
    >
      wind
    </span>
    <input
      defaultValue={0.5}
      max="1"
      min="0"
      onChange={[Function]}
      onClick={[Function]}
      step="0.01"
      type="range"
    />
  </div>
  <div
    className="youtube-video"
    id="video-wind"
  />
</div>
`;

exports[`<SoundItem/> youtube render test-id 1`] = `
<div
  className="item waves-effect waves-block paused youtube-stream"
  onClick={[Function]}
  role="button"
  style={
    Object {
      "color": "#121212",
    }
  }
  tabIndex={-1}
>
  <div
    className="inner"
  >
    <img
      alt="wind"
      src="wind"
    />
    <div
      className="actions dark"
    >
      <div>
        <a
          href="wind"
          onClick={[Function]}
          rel="noopener noreferrer"
          target="_blank"
        >
          <i
            className="icon-share"
          />
        </a>
      </div>
      
      <div
        onClick={[Function]}
        role="presentation"
      >
        <i
          className="icon-delete"
        />
      </div>
    </div>
    <span
      className="title"
    >
      wind
    </span>
    <input
      defaultValue={0.5}
      max="1"
      min="0"
      onChange={[Function]}
      onClick={[Function]}
      step="0.01"
      type="range"
    />
  </div>
  <div
    className="youtube-video"
    id="video-7ccQyyCLtx8"
  />
</div>
`;
